packages/next/README.md

```diff
@@ -908,6 +912,51 @@ class MyLink extends React.Component {
 export default withRouter(MyLink)
 ```
 
+### API routes
+
+<details>
+  <summary><b>Examples</b></summary>
+  <ul>
+    <li><a href="/examples/api-routes">Basic API routes</a></li>
+    <li><a href="/examples/api-routes-micro">API routes with micro</a></li>
+  </ul>
+</details>
+
+API routes provides straightforward solution to build your `API` with `Next.js`. Start by creating `api` folder inside `pages` folder. Every file inside `api` folder is mapped to `/api` route. For example `api/posts.js` will be mapped to `/api/posts`.
+
+```js
+export default (req, res) => {
+  res.setHeader('Content-Type', 'application/json')
+  res.statusCode = 200
+  res.end(JSON.stringify({ name: 'Nextjs' }))
+}
+```
+
+**Note: API routes pages are compiled just for the server. So there is no overhead inside the client bundle. Every route is its own separated bundle, so there is no problem with big bundles and slow bootup.**
+
+
+#### API middlewares
+
+API routes provides built in `middlewares` which parse incoming `req` event. These middlewares are:
+
+- `req.cookies` - an object containing the cookies sent by the request - default `{}` 
+- `req.query` - an object containing the request's [query string](https://en.wikipedia.org/wiki/Query_string) - default `{}`
+- `req.body` - an object containing parsed body by `content-type`, or `null` if no body is sent
+
+#### Helper functions
+
+We're providing a set of `Express.js`-like methods to improve the developer experience and increase the speed of creating a new `API` endpoints.
+
+```js
+export default (req, res) => {
+  res.status(200).json({ name: 'Nextjs' })
+}
+```
+
+- `res.status(code)` - a function to set status code send with the response where `code` must be valid [HTTP status code](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes)
+- `res.json(json)` - a function to send `JSON` response where `json` is the `JSON` object to send
```

Comment:
- `res.json(json)` - Sends a `JSON` response where `json` must be a valid `JSON` object